### DISCUSSION TITLE
Slack V1 OAuth Provider Deprecated in favour of Slack (OIDC)

### DISCUSSION THREAD

**Discussion by J0** [supabase/supabase#30772] [Changelog]
Dear Users,

Slack has [deprecated their existing v1 API](https://api.slack.com/changelog/2024-04-discontinuing-new-creation-of-classic-slack-apps-and-custom-bots) and will fully sunset the API over the [next two years](https://api.slack.com/changelog/2024-09-legacy-custom-bots-classic-apps-deprecation). This means that Slack Applications using the v1 `Slack (Deprecated)` provider will stop functioning if no action is taken.  In turn, Supabase Auth users using the `Slack (Deprecated)` provider will also not be able to sign in via the Slack OAuth provider if no action is taken.

<img width="854" alt="image" src="https://github.com/user-attachments/assets/eca053a4-72ac-4b4f-aca4-1ef97362a2ed">

We’ve introduced a new replacement OAuth provider named “Slack (OIDC)” which supports the latest Slack API and will send out an email to affected users next week with instructions on how to migrate.

To migrate your authentication system, you'll need to:

1. [Create a new Slack Application](https://api.slack.com/quickstart) in your Slack dashboard
2. Update your credentials in the Supabase Dashboard by switching from the deprecated `Slack (Deprecated)`  provider to the new `Slack (OIDC)` provider. You can find the provider in your project under [Authentication > Providers > Slack (Deprecated)](https://supabase.com/dashboard/project/_/auth/providers)
3. Update your code to use the new provider in all OAuth calls, like this:

```jsx
const { data, error } = await supabase.auth.signInWithOAuth({
    provider: 'slack_oidc',
  })
```
4. If you are using the Supabase CLI , you will also need to update your `config.toml` to use `[auth.external.slack_oidc]` in place of `[auth.external.slack_oidc]`. See the [local development docs](https://supabase.com/docs/guides/local-development/overview#use-auth-locally) for a detailed example.

Refer to our documentation for [further details](https://supabase.com/docs/guides/auth/social-login/auth-slack)

Further configuration of the deprecated Slack provider will not be possible past **15th Jan 2025** as we will remove `Slack` provider from the dashboard then.  If you need more time to access the configuration page , reach out to us [via a support ticket](https://supabase.com/dashboard/support/new). 

Slack will terminate support for Legacy bots and Classic Apps on March 31, 2025 and March 31, 2026 respectively.

Please reach out via support or on the thread if there are any questions or concerns.

**Comment by b4lk0n** [comment]
Though the configuration of the deprecated provider won’t be possible, will it still be working past Jan 15?
